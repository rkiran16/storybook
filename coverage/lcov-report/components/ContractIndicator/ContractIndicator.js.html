
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components/ContractIndicator/ContractIndicator.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">components/ContractIndicator</a> ContractIndicator.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>4/4</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>3/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>4/4</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { html } from 'lit-html';
&nbsp;
const contractIcon = () =&gt; {
   return html`
    &lt;span class="position-relative me-2 d-flex align-items-center"&gt;
      &lt;label class="position-absolute top-50 start-50 sub-title-4 translate-middle text-white lh-1"&gt;8&lt;/label&gt;
      &lt;svg role="img" aria-labelledby="count" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"&gt;
        &lt;title id="count"&gt;8&lt;/title&gt;
        &lt;path fill="currentColor" d="m25 12.955-2.662-3.044.371-4.025-3.938-.895L16.709 1.5 13 3.093 9.291 1.5 7.229 4.98l-3.938.884.371 4.036L1 12.955 3.662 16l-.371 4.036 3.938.895 2.062 3.48L13 22.805l3.709 1.595 2.062-3.48 3.938-.895L22.338 16Z"/&gt;
      &lt;/svg&gt;
    &lt;/span&gt;
   `
}
&nbsp;
&nbsp;
export const ContractIndicator = ({variation = <span class="branch-0 cbranch-no" title="branch not covered" >'secondary',</span> label = <span class="branch-0 cbranch-no" title="branch not covered" >"CONTRACT 33590890",</span> hasIcon = true}) =&gt; {
   return html `
      &lt;span class=${`d-flex align-items-center text-${variation}`}&gt;
        ${hasIcon ? contractIcon() : '' } 
        &lt;strong class="pt-1"&gt;${label}&lt;/strong&gt;
      &lt;/span&gt;
   `
}</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2022-09-17T00:07:40.092Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    