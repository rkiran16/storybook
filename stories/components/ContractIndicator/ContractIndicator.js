import { html } from 'lit-html';

const contractIcon = () => {
   return html`
    <span class="position-relative me-2 d-flex align-items-center">
      <label class="position-absolute top-50 start-50 sub-title-4 translate-middle text-white lh-1">8</label>
      <svg role="img" aria-labelledby="count" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
        <title id="count">8</title>
        <path fill="currentColor" d="m25 12.955-2.662-3.044.371-4.025-3.938-.895L16.709 1.5 13 3.093 9.291 1.5 7.229 4.98l-3.938.884.371 4.036L1 12.955 3.662 16l-.371 4.036 3.938.895 2.062 3.48L13 22.805l3.709 1.595 2.062-3.48 3.938-.895L22.338 16Z"/>
      </svg>
    </span>
   `
}


export const ContractIndicator = ({variation = 'secondary', label = "CONTRACT 33590890", hasIcon = true}) => {
   return html `
      <span class=${`d-flex align-items-center text-${variation}`}>
        ${hasIcon ? contractIcon() : '' } 
        <strong class="pt-1">${label}</strong>
      </span>
   `
}