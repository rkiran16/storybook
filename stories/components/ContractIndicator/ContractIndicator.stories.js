import { ContractIndicator } from "./ContractIndicator";
import { html } from 'lit-html';


export default {
  title: "Components/ContractIndicator",
  parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6305428474d6e412c3a078a3",
  },
  argTypes: {
     variation: {
        name: "Color Variations",
        control: { type: 'select'},
        description: "Assign Color Variations",
        defaultValue: "info",
        options: ['dark', 'primary', 'info', 'secondary', 'success', 'danger', 'warning'],
        table: {
          type: { summary: 'select' },
          defaultValue: { summary: "info" },
        },
     }
  }
}

const Template = args => ContractIndicator(args);
const TemplateWTitle = args => {
   return html `
    <div>
      ${ContractIndicator(args)}
      <h1 class="ms-n1 mt-2">Heading 1</h1>
    </div>
   `
}

export const Standard = Template.bind({});
Standard.args = {
   label: "CONTRACT 33590890"
}

export const EyeBrow = Template.bind({});

EyeBrow.args = {
  label: 'NEW!',
  hasIcon: false
}

export const EyeBrow_W_Title = TemplateWTitle.bind({});
EyeBrow_W_Title.args = {
   label: 'NEW!',
   hasIcon: false
}