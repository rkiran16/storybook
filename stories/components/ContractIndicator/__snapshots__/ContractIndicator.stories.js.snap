// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/ContractIndicator Eye Brow 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <span class=",
    ">
        ",
    " 
        <strong class=\\"pt-1\\">",
    "</strong>
      </span>
   ",
  ],
  "values": Array [
    "d-flex align-items-center text-info",
    "",
    "NEW!",
  ],
}
`;

exports[`Storyshots Components/ContractIndicator Eye Brow W Title 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div>
      ",
    "
      <h1 class=\\"ms-n1 mt-2\\">Heading 1</h1>
    </div>
   ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <span class=",
        ">
        ",
        " 
        <strong class=\\"pt-1\\">",
        "</strong>
      </span>
   ",
      ],
      "values": Array [
        "d-flex align-items-center text-info",
        "",
        "NEW!",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/ContractIndicator Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
      <span class=",
    ">
        ",
    " 
        <strong class=\\"pt-1\\">",
    "</strong>
      </span>
   ",
  ],
  "values": Array [
    "d-flex align-items-center text-info",
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <span class=\\"position-relative me-2 d-flex align-items-center\\">
      <label class=\\"position-absolute top-50 start-50 sub-title-4 translate-middle text-white lh-1\\">8</label>
      <svg role=\\"img\\" aria-labelledby=\\"count\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">
        <title id=\\"count\\">8</title>
        <path fill=\\"currentColor\\" d=\\"m25 12.955-2.662-3.044.371-4.025-3.938-.895L16.709 1.5 13 3.093 9.291 1.5 7.229 4.98l-3.938.884.371 4.036L1 12.955 3.662 16l-.371 4.036 3.938.895 2.062 3.48L13 22.805l3.709 1.595 2.062-3.48 3.938-.895L22.338 16Z\\"/>
      </svg>
    </span>
   ",
      ],
      "values": Array [],
    },
    "CONTRACT 33590890",
  ],
}
`;
